### PR TITLE
add go todo app task

### PR DESCRIPTION
**Pull request by gusdecool**
issue: https://github.com/bali-gophers/backpack/issues/3

sample to-do-app using go language

### CODE REVIEW

File: examples/to-do-app/db/repository/task/task.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	db.Find(&tasks)
+
+	return tasks, nil
+}
+
+func GetOneById(id int) (model.Task, error) {
+	db, err := connector.Connect()
+
+	defer db.Close()
```
Comment (rakateja) [inline examples/to-do-app/db/repository/task/task.go]: same like above, please check `err` first before use `db`

File: examples/to-do-app/db/repository/task/task.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		return task, errors.New(fmt.Sprintf("can't find task with id %d", id))
+	}
+
+	return task, nil
+}
+
+func Create(task *model.Task) (*model.Task, error) {
+	db, err := connector.Connect()
+
+	defer db.Close()
```
Comment (rakateja) [inline examples/to-do-app/db/repository/task/task.go]: same like above, please check `err` first before use `db`

File: examples/to-do-app/db/repository/task/task.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	db.Create(&task)
+
+	return task, nil
+}
+
+func Update(task *model.Task) (*model.Task, error) {
+	db, err := connector.Connect()
+
+	defer db.Close()
```
Comment (rakateja) [inline examples/to-do-app/db/repository/task/task.go]: same like above, please check `err` first before use `db`

File: examples/to-do-app/db/repository/task/task.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		return task, err
+	}
+
+	return task, nil
+}
+
+func Delete(task *model.Task) error {
+	db, err := connector.Connect()
+
+	defer db.Close()
```
Comment (rakateja) [inline examples/to-do-app/db/repository/task/task.go]: same like above, please check `err` first before use `db`

File: examples/to-do-app/http/controller/task/task.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	if err != nil {
+		utility.HandleErrorResponse(err, response)
+		return
+	}
+
+	utility.HandleSuccessResponse(tasksByte, response)
+}
+
+func Create(response http.ResponseWriter, request *http.Request) {
+	taskModel, err := decode(request)
```
Comment (rakateja) [inline examples/to-do-app/http/controller/task/task.go]: before use taskModel, please check err first

File: examples/to-do-app/db/connector/connector.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/mysql"
+)
+
+func Connect() (*gorm.DB, error) {
+	db, err := gorm.Open("mysql", "root@/go-todo-app?charset=utf8&parseTime=True&loc=Local")
+
+	db.SingularTable(true)
```
[the remark this answers is not in the document]
Comment (gusdecool) [reply examples/to-do-app/db/connector/connector.go]: i've added `err` check here https://github.com/bali-gophers/backpack/pull/7/files#diff-d15b797a03ae0fe42442c4308adfd7d4R11

File: examples/to-do-app/http/controller/task/task.go (hunk 5 above)
Comment (gusdecool) [reply examples/to-do-app/http/controller/task/task.go]: I've added `err` check here https://github.com/bali-gophers/backpack/pull/7/files#diff-47328041993fa3e8acb14f43b8f08bb1R35

**Comment by gusdecool** [comment]
Hi @rakateja , PR updated :)

**Pull request merged**